wp-content/themes/twentyseventeen-wp20/footer.php

```diff
@@ -1,47 +1,72 @@
-			<div class="wrap wrap-wide">
+			<div class="wp20-pre-footer">
+				<div class="wrap wrap-wide">
+					<p>
+						<?php 
+							if ( is_page( 'swag' ) ) {
+								printf(
+									wp_kses_data( __( 'Check out the <a href="%s">WordPress Swag Store</a> to get your hands on new, limited edition WP20 swag.', 'wp20' ) ),
+									'https://mercantile.wordpress.org/product-category/wordpress-20/'
+								);
+							} elseif ( is_page( 'whats-on' )) {
+								printf(
+									wp_kses_post( __( '<strong>Don’t see your city?</strong><br> Get in touch with <a href="%s">your local group</a>, or <a href="%s">organize a group in your town</a>.', 'wp20' ) ),
+									'https://www.meetup.com/pro/wordpress/',
+									'https://make.wordpress.org/community/handbook/meetup-organizer/welcome/'
+								);
+							} else {
+								echo wp_kses_post( __( 'Follow <strong>#WP20</strong> on social for latest updates, or be part of celebration by posting.', 'wp20' ) );
```

Comment:
Is the escaping/translations right here? I'm still not very clear when to use `wp_kses_data`, `wp_kses_post`, `esc_html_e`, etc...

Reply:
In general I use `esc_*` by default, but sometimes you want to include HTML in the output, like the `<strong>` tag here. In those cases I use the `kses_` functions, and I pick the most restrictive one that allows everything in the string. `post` is what gets applied to posts, and the poorly-named `data` is what gets applied to comments by default. You can [see the exact tags in `wp-includes/kses.php`](https://github.com/WordPress/WordPress/blob/1764cee3d1a7772992d04222dfcbd18084754eb6/wp-includes/kses.php#L68-L724)

The reason escaping/sanitization is used here at all is because a hacker could compromise a translator's account, and then insert malicious JavaScript etc.

Reply:
Thanks that's really helpful. I had to use `post` to get the `<br>` tag allowed. `<strong>` worked with both.